### PR TITLE
Added in locked perks to the sockets shown in the armour 2 mod picker

### PR DESCRIPTION
**Pull request by ryan-rushton**
The locked perks will now be visible in the new mod picker

image
image

### CODE REVIEW

**Comment by bhollis** [comment]
Nice! Can we remove the border from perks and display them with a circle background instead?

**Comment by ryan-rushton** [comment]
Check that out now, I also updated the pics in the initial comment.

Just as an FYI the old perks currently show any perk selected in perk picker but the plan is eventually to trim that down to pre armour 2.0 mods, perks and burns only. Obviously that might change depending on how the rest of this goes.

File: src/app/loadout-builder/generated-sets/GeneratedSetItem.tsx [hunk 1]
```diff
@@ -120,10 +136,26 @@ export default function GeneratedSetItem({
...
+        <div className={styles.lockedSockets}>
+          {Boolean(lockedMods?.length) && (
+            <div className={`lockedItems ${styles.lockedSocketsRow}`}>
+              {lockedMods?.map((mod) => (
+                <SocketDetailsMod key={mod.mod.hash} itemDef={mod.mod} defs={defs} />
+              ))}
+            </div>
+          )}
+          {Boolean(lockedOldMods.length) && (
+            <div className={`lockedItems ${styles.lockedSocketsRow}`}>
```
Comment (bhollis) [inline src/app/loadout-builder/generated-sets/GeneratedSetItem.tsx:148]: We've been using `clsx` for this sort of thing.

Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/GeneratedSetItem.tsx:148]: Cool, I will clean it up in the next one

**Pull request merged**